src/admin/resolvers/queries/CollectionStory.integration.ts

```diff
@@ -0,0 +1,98 @@
+import { expect } from 'chai';
+import {
+  clear as clearDb,
+  createAuthorHelper,
+  createCollectionHelper,
+  createCollectionStoryHelper,
+  sortCollectionStoryAuthors,
+  getServerWithMockedHeaders,
+} from '../../../test/helpers';
+import { db } from '../../../test/admin-server';
+import { COLLECTION_CURATOR_FULL } from '../../../shared/constants';
+import { GET_COLLECTION_STORY } from './sample-queries.gql';
+
+describe('queries: CollectionStory', () => {
+  const headers = {
+    name: 'Test User',
+    username: 'test.user@test.com',
+    // default to full access - auth tests occur in a separate test file
+    groups: `group1,group2,${COLLECTION_CURATOR_FULL}`,
+  };
+
+  // note that calling `executeOperation` on this server does not require
+  // calling `server.start()`
+  const server = getServerWithMockedHeaders(headers);
+
+  beforeEach(async () => {
+    await clearDb(db);
+  });
+
+  afterAll(async () => {
+    await db.$disconnect();
+  });
+
+  describe('getCollectionStory', () => {
+    let story;
+
+    beforeEach(async () => {
+      const author = await createAuthorHelper(db, 'donny');
+      const collection = await createCollectionHelper(db, {
+        title: 'test me',
+        author,
+      });
+
+      story = await createCollectionStoryHelper(db, {
+        collectionId: collection.id,
+        url: 'https://getpocket.com',
+        title: 'a story',
+        excerpt: 'this is a story, all about how...',
+        imageUrl: 'https://some.image',
+        authors: [{ name: 'donny', sortOrder: 0 }],
+        publisher: 'the verge',
+        fromPartner: false,
+      });
+    });
+
+    it('should retrieve a collection story with authors', async () => {
+      const { data } = await server.executeOperation({
+        query: GET_COLLECTION_STORY,
+        variables: {
+          externalId: story.externalId,
+        },
+      });
+
+      const retrieved = data?.getCollectionStory;
+
+      expect(retrieved.title).to.equal('a story');
+      expect(retrieved.authors.length).to.be.greaterThan(0);
+    });
+
+    it('should retrieve a collection story with authors sorted correctly', async () => {
+      const { data } = await server.executeOperation({
+        query: GET_COLLECTION_STORY,
+        variables: {
+          externalId: story.externalId,
+        },
+      });
+
+      const retrieved = data?.getCollectionStory;
+
+      // the default sort returned from prisma should match our expected
+      // manual sort
+      expect(retrieved.authors).to.equal(
+        sortCollectionStoryAuthors(retrieved.authors)
+      );
+    });
+
+    it('should fail to retrieve a collection story for an unknown externalID', async () => {
+      const { data } = await server.executeOperation({
+        query: GET_COLLECTION_STORY,
+        variables: {
+          externalId: story.externalId + 'typo',
+        },
+      });
+
+      expect(data.getCollectionStory).to.be.null;
```

Comment:
Are we throwing a NOT_FOUND error here that we could also expect? Not blocking but I only ask in relation to the effort to standardize error messaging for clients.

Reply:
@kkyeboah I think we don't use custom error codes from `apollo-server-errors` and `apollo-utils` in this repository yet. Maybe the new authentication-related one only? Would be a nice refactor/cleanup ticket to update the other exceptions into semantically meaningful ones rather than just internal server errors.

Reply:
i would love application consistency in how we handle different non-happy path scenarios. @kkyeboah do you have any documentation from infra - list api or other?